src/main/java/com/thebois/models/world/generation/TerrainGenerator.java

```diff
@@ -0,0 +1,66 @@
+package com.thebois.models.world.generation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.thebois.models.world.generation.noises.INoise;
+import com.thebois.models.world.generation.noises.NoiseFactory;
+import com.thebois.models.world.terrains.ITerrain;
+import com.thebois.models.world.terrains.TerrainFactory;
+import com.thebois.models.world.terrains.TerrainType;
+import com.thebois.utils.MatrixUtils;
+
+/**
+ * Generator used to generate terrain tiles for the world.
+ */
+public class TerrainGenerator extends AbstractGenerator {
+
+    private final Map<TerrainType, INoise> terrainNoise;
+
+    /**
+     * Instantiate a Terrain Generator with pre-made settings used for generating terrain.
+     *
+     * @param worldSize The world size of the world.
+     * @param seed      The seed used to generate the world.
+     */
+    public TerrainGenerator(final int worldSize, final int seed) {
+        super(worldSize, seed);
+        terrainNoise = new HashMap<>();
+        terrainNoise.put(TerrainType.SAND, NoiseFactory.createLargeChunksNoise());
+        terrainNoise.put(TerrainType.GRASS, NoiseFactory.createFillMapNoise());
+        terrainNoise.put(TerrainType.DIRT, NoiseFactory.createLargeChunksNoise());
```

Comment:
A way to solve that^ issue would be having SAND_DESERT and SAND_BEACH, and in real life they're usually two different types of sand, too.

Reply:
Would it really? How would the TerrainType know when it should cover the world in mostly sand and not dirt and grass?

"SAND_DESERT and SAND_BEACH" would allow us to have sand spawn somewhere where there isn't water if we want that

Reply:
It's the Generator that decides whether or not to generate a desert environment or not. If it does, then it uses SAND_DESERT. If it doesn't, then it uses SAND_BEACH. 

Are you suggesting that we'd want to have both beaches and deserts at the same time? If so, then it still works, the generator just uses both sand types at the same time. The generator is responsible for how they overlap and which one takes precedence.

Reply:
Well, even then, it would work. The user selects that they want to play in a desert, and so we set the generator to `DesertGenerator` instead of the usual `TemperateGenerator`.

Reply:
For example if sand have the `largeChunkNoise `assoicated with it, if we create a desert then we would probably have the `fillMapNoise `instead no? Wait, damn, it would need to change its threshold as well 😡

Reply:
Exactly, we wouldn't ever change the noise for a TerrainType. We'd have multiple TerrainTypes instead. 

If we wanted to create a desert or a tropical environment instead of the default, we'd have:
1) `TerrainType.SAND_DESERT` and have a `DesertGenerator` that uses it
2) The tropical terrain types, and a `TropicalGenerator` that uses it

Reply:
I guess we could create a enum value for all kind of terrains, i.e DESERT_SAND, TEMPERATE_SAND, and then the DesertGenerator have to chose the right enum.

Reply:
Sounds like you're still thinking of only having a single SAND terrain type, that problem is solved if we have multiple different types of sand-terrains.

Reply:
Yeah, isn't it super reasonable actually to have a different `TerrainType` for each different type of terrain we want to generate, even though they might be variations of the same material, e.g. sand.

Reply:
Doesn't necessarily have to cause duplicated code, but you're right that it would need to be handled differently in the views.

I'm imagining it'd be quite easy to have a hashmap with <List<TerrainType>, Texture>, and so you would just add all different types of sand into the same texture-terraintypes-list and the view could then just look it up and see that it should use the same texture for all sands.